plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3AsyncService.java

```diff
@@ -131,8 +132,9 @@ public AmazonAsyncS3Reference client(
                 return existing;
             }
 
+            String asyncHttpClientType = S3Repository.S3_ASYNC_HTTP_CLIENT_TYPE.get(repositoryMetadata.settings());
```

Comment:
It appears that client caching will be buggy where the setting `s3_async_client_type` may get ignore if the same S3Client setting exists already. Lets validate the same. 
https://github.com/opensearch-project/OpenSearch/blob/30519a3826eb366e66676494271ad59e0e920cb2/plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3AsyncService.java#L115-L141

Reply:
Thanks for Pointing this. Have added support for caching Netty/CRT based for registered snapshots and updates.